tests/validation/tests/v3_api/test_node_annotation.py

```diff
@@ -0,0 +1,554 @@
+         PROJECT_READ_ONLY]
+
+
+def test_node_annotation_add():
+    annotation_key = random_name()
+    annotation_value = random_name()
+    # get node details
+    client, node = \
+        get_node_details(cluster_detail["cluster"], cluster_detail["client"])
+
+    # add annotation through API
+    node_annotations = node.annotations.data_dict()
+    node_annotations[annotation_key] = annotation_value
+    client.update(node, annotations=node_annotations)
+    time.sleep(2)
+    # annotation should be added
+    validate_annotation_set_on_node(client, node,
+                                    annotation_key,
+                                    annotation_value)
+
+    # delete annotation
+    del node_annotations[annotation_key]
+    client.update(node, annotations=node_annotations)
+
+
+def test_node_annotation_edit():
+    annotation_key = random_name()
+    annotation_value = random_name()
+    # get node details
+    client, node = \
+        get_node_details(cluster_detail["cluster"], cluster_detail["client"])
+
+    # add annotation through API
+    node_annotations = node.annotations.data_dict()
+    node_annotations[annotation_key] = annotation_value
+    client.update(node, annotations=node_annotations)
+    time.sleep(2)
+
+    # annotation should be added
+    node = client.reload(node)
+    validate_annotation_set_on_node(client, node,
+                                    annotation_key,
+                                    annotation_value)
+
+    # edit annotation through API
+    node = client.reload(node)
+    node_annotations = node.annotations.data_dict()
+    new_value = random_name()
+    node_annotations[annotation_key] = new_value
+    client.update(node, annotations=node_annotations)
+    node = client.reload(node)
+    time.sleep(2)
+
+    validate_annotation_set_on_node(client, node,
+                                    annotation_key,
+                                    new_value)
+
+    # delete annotation
+    del node_annotations[annotation_key]
+    client.update(node, annotations=node_annotations)
+
+
+def test_node_annotation_delete():
+    annotation_key = random_name()
+    annotation_value = random_name()
+    # get node details
+    client, node = \
+        get_node_details(cluster_detail["cluster"], cluster_detail["client"])
+
+    # add annotation on node
+    node_annotations = node.annotations.data_dict()
+    node_annotations[annotation_key] = annotation_value
+    client.update(node, annotations=node_annotations)
+    time.sleep(2)
+
+    # annotation should be added
+    validate_annotation_set_on_node(client, node,
+                                    annotation_key,
+                                    annotation_value)
+
+    # delete annotation
+    del node_annotations[annotation_key]
+    client.update(node, annotations=node_annotations)
+    time.sleep(2)
+
+    # annotation should be deleted
+    validate_annotation_deleted_on_node(client, node, annotation_key)
```

Comment:
Can we add and delete multiple annotations ?

Reply:
Done